JASP-Engine/JASP/R/ancova.R

```diff
@@ -3176,30 +3176,14 @@ Ancova <- function(dataset=NULL, options, perform="run", callback=function(...)
     xhigh <- max(pretty(xVar))
     xticks <- pretty(c(xlow, xhigh))
     
-    # format x labels
-    xLabs <- vector("character", length(xticks))
-    for (i in seq_along(xticks)) {
-      if (xticks[i] < 10^6) {
-        xLabs[i] <- format(xticks[i], digits= 3, scientific = FALSE)
-      } else {
-        xLabs[i] <- format(xticks[i], digits= 3, scientific = TRUE)
-      }
-    }
-    
-    # Format y ticks
+      # Format y ticks
     ylow <- min(pretty(yVar))
     yhigh <- max(pretty(yVar))        
     yticks <- pretty(c(ylow, yhigh))
     
-    # format y labels
-    yLabs <- vector("character", length(yticks))
-    for (i in seq_along(yticks)) {
-      if (yticks[i] < 10^6) {
-        yLabs[i] <- format(yticks[i], digits= 3, scientific = FALSE)
-      } else {
-        yLabs[i] <- format(yticks[i], digits= 3, scientific = TRUE)
-      }
-    }
+    # format axes labels
+    xLabs <- JASPgraphs::axesLabeller(xticks)
+    yLabs <- JASPgraphs::axesLabeller(yticks)
     
     p <- JASPgraphs::drawAxis(xName = "Theoretical Quantiles", yName = "Standardized Residuals", xBreaks = xticks, yBreaks = xticks, yLabels = xLabs, xLabels = xLabs, force = TRUE)
```

Comment:
Can't `JASPgraphs::drawAxis` take care of `JASPgraphs::axesLabeller(xticks)`? Are there situations we do not want this?

Reply:
I actually want to get rid of `drawAxis` and bring the JASPgraphs API closer to the usual ggplot2 code. I would prefer something like:
```r
ggplot2::ggplot(data = dat, mapping = ggplot2::aes(...)) + 
  # ggplot2::geom_point() but with our defaults
  JASPgraph::geom_point() +
  # mimics ggplot2::scale_x_continuous and has an argument called labels, which could default to axesLabeller
  JASPgraphs::scale_x_continuous(xName = ...) + 
  # creates the lines on the axes (mimic's base r, bty = "n")
  JASPgraphs::geom_rangeframe() + 
  # add the JASP theme
  JASPgraphs::themeJasp()
```
That's something for the future though, when I'm happy about how `JASPgraphs::geom_rangeframe()` works. I also want the API to be stable before I start to deprecate things and rewrite plots to a common style.

Reply:
But.. then my question stands. Are there situations where we don't want the `axesLabeller` functionality? Because until you've done a rewrite it'll be something people will have to think of.

Reply:
well `axesLabeller` doesn't quite work for factors.
```r
JASPgraphs::axesLabeller(factor(letters[1:3]))
[1] "1" "2" "3"
attr(,"levels")
[1] "a" "b" "c"
```

Aside from that I think it can be used for everything, so there is no real reason not to make it the default (sorry I missed that the first time). I'll change `drawAxis()` so that it uses `axesLabeller` automatically if xBreaks or yBreaks are supplied.